tests/suites/test_suite_pem.data

```diff
@@ -20,22 +20,25 @@ PEM read (unencrypted, valid)
 mbedtls_pem_read_buffer:"^":"$":"^\nTWJlZCBUTFM=\n$":"":0:"4d62656420544c53"
 
 PEM read (DES-EDE3-CBC + invalid iv)
+depends_on:MBEDTLS_HAS_ALG_MD5_VIA_MD_OR_PSA_BASED_ON_USE_PSA:MBEDTLS_CIPHER_MODE_CBC:MBEDTLS_DES_C
 mbedtls_pem_read_buffer:"^":"$":"^\nProc-Type\: 4,ENCRYPTED\nDEK-Info\: DES-EDE3-CBC,00$":"pwd":MBEDTLS_ERR_PEM_INVALID_ENC_IV:""
 
 PEM read (DES-CBC + invalid iv)
+depends_on:MBEDTLS_HAS_ALG_MD5_VIA_MD_OR_PSA_BASED_ON_USE_PSA:MBEDTLS_CIPHER_MODE_CBC:MBEDTLS_DES_C
 mbedtls_pem_read_buffer:"^":"$":"^\nProc-Type\: 4,ENCRYPTED\nDEK-Info\: DES-CBC,00$":"pwd":MBEDTLS_ERR_PEM_INVALID_ENC_IV:""
 
 PEM read (unknown encryption algorithm)
+depends_on:MBEDTLS_HAS_ALG_MD5_VIA_MD_OR_PSA_BASED_ON_USE_PSA:MBEDTLS_CMAC_C
```

Comment:
Why `MBEDTLS_CMAC_C`? It seems that this is a new dependency.

Reply:
There was function level dependency to `MBEDTLS_CIPHER_MODE_CBC` and now it is only for `CBC` cases (below). But without `MBEDTLS_CIPHER_MODE_CBC` this test was executed and was failing. While debugging I decided to add `MBEDTLS_CMAC_C` as it was more suitable.

Reply:
This doesn't make sense: why a dependency on a completely unrelated algorithm? PEM doesn't use CMAC directly or indirectly. Maybe a dependency is missing but CMAC can't be right. Looking at the code, I think there's a missing dependency on `MBEDTLS_AES_C` (or more precisely `MBEDTLS_DEC_C || MBEDTLS_AES_C`, but we don't really care about DES-only configurations), because without that the PEM module doesn't even try to parse encryption.

Reply:
Will check again.

Reply:
Removed this dependency and checked locally `default, full, test_psa_crypto_config_accel_hash_use_psa, test_psa_crypto_config_accel_hash` configurations and everything is ok. I also can't find related CI failure in the history as it doesn't go that far.
In this case I will remove this and let's see what CI says.

Reply:
Yup, CI has failed now. The following component:

```
^^^^test_when_no_ciphersuites_have_mac: test: !MBEDTLS_SSL_SOME_MODES_USE_MAC: make test -> 2^^^^


PEM read (unknown encryption algorithm) ........................... FAILED
  ret == res
  at line 51, suites/test_suite_pem.function

From the component:
    scripts/config.py unset MBEDTLS_CIPHER_NULL_CIPHER
    scripts/config.py unset MBEDTLS_CIPHER_MODE_CBC
    scripts/config.py unset MBEDTLS_CMAC_C
```

Reply:
The dependency is on `MBEDTLS_CIPHER_MODE_CBC` (you can confirm by doing a build with default config minus `MBEDTLS_CIPHER_MODE_CBC`). The test case only fails because `mbedtls_pem_read_buffer` returns a different error (`MBEDTLS_ERR_PEM_FEATURE_UNAVAILABLE`).

To be very precise in the tests, we should duplicate the test cases for this function and make them expect `MBEDTLS_ERR_PEM_FEATURE_UNAVAILABLE` when the encryption mechanism isn't supported. As a quick fix, we can make the test case depend on `MBEDTLS_CIPHER_MODE_CBC`. (It actually depends on more than that, but there are many missing dependencies on cipher features throughout the test suites and fixing them is out of scope here.)

Reply:
Thanks for clarification. I have modified the last commit and changed dependency from `MBEDTLS_CMAC_C to MBEDTLS_CIPHER_MODE_CBC`.

Reply:
By the way, https://github.com/Mbed-TLS/mbedtls/issues/1892 tracks the lack of test builds with variations on supported cipher features. No point in fixing preexisting individual cases until we have a way to test.